components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2439,6 +2440,33 @@ public static OauthTokenIssuer getTokenIssuer(String accessToken) throws Identit
         }
         return oauthTokenIssuer;
     }
+
+    /**
+     * Publish event on token generation error.
+     *
+     * @param exception Exception occurred.
+     * @param params Additional parameters.
+     */
+    public static void triggerOnTokenExceptionListeners(Throwable exception, Map<String, Object> params) {
+
+        try {
+            OAuthEventInterceptor oAuthEventInterceptorProxy = OAuthComponentServiceHolder.getInstance()
+                                                                                          .getOAuthEventInterceptorProxy();
+
+            if (oAuthEventInterceptorProxy != null) {
+                try {
+                    oAuthEventInterceptorProxy.onTokenIssueException(exception, params);
+                } catch (IdentityOAuth2Exception e) {
+                    log.error("Error while invoking OAuthEventInterceptor for onTokenIssueException", e);
+                }
+            }
+        } catch (Throwable e) {
+            // Catching a throwable as we do no need to interrupt the code flow since these are logging purposes.
```

Comment:
For the public branch, shall we add an error log or shall we make this log error type ?. Otherwise due to this there might be logs missing without noticiing